### PR TITLE
update codeowners

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

discussed with @tchaton @SeanNaren 
also fixes @williamFalcon approval fro changing this file

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -5,23 +5,42 @@
...
+pytorch_lightning/core/lightning.py           @williamfalcon @tchaton
+pytorch_lightning/trainer/training_loop.py    @tchaton
+pytorch_lightning/trainer/evaluation_loop.py  @tchaton
 
+# Connectors
+pytorch_lightning/trainer/connectors/logger_connector/*      @tchaton
 
 # accelerators
-/pytorch_lightning/accelerators/*  @williamfalcon
+pytorch_lightning/accelerators/*              @williamfalcon @tchaton @SeanNaren
```
Comment (tchaton) [inline .github/CODEOWNERS:28]: Can you add @awaelchli  and @justusschock  ?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -5,23 +5,42 @@
...
 
 # owners
-/pytorch_lightning/.github/CODEOWNERS @williamfalcon
+.github/CODEOWNERS   @williamfalcon
+# main
+README.md            @williamfalcon
+
```
Comment (Borda) [inline .github/CODEOWNERS:34]:
```suggestion
# installation
README.md            @borda @williamfalcon

```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -5,23 +5,43 @@
...
 
 # accelerators
-/pytorch_lightning/accelerators/*  @williamfalcon
+pytorch_lightning/accelerators/*              @williamfalcon @tchaton @SeanNaren
 
 # owners
-/pytorch_lightning/.github/CODEOWNERS @williamfalcon
+.github/CODEOWNERS   @williamfalcon
+# main
+README.md            @williamfalcon
```
Comment (Borda) [inline .github/CODEOWNERS:33]:
```suggestion
README.md            @williamfalcon @edenlightning
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -5,23 +5,43 @@
...
+pytorch_lightning/callbacks/base.py           @williamfalcon
+pytorch_lightning/core/datamodule.py          @williamfalcon
+pytorch_lightning/trainer/trainer.py          @williamfalcon @tchaton
+pytorch_lightning/core/hooks.py               @williamfalcon
+pytorch_lightning/core/lightning.py           @williamfalcon @tchaton
+pytorch_lightning/trainer/training_loop.py    @tchaton
+pytorch_lightning/trainer/evaluation_loop.py  @tchaton
 
+# Connectors
+pytorch_lightning/trainer/connectors/logger_connector/*      @tchaton
```
Comment (Borda) [inline .github/CODEOWNERS:25]:
```suggestion
pytorch_lightning/trainer/connectors/logger_connector/*      @tchaton @SeanNaren
```

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -5,23 +5,43 @@
...
-/pytorch_lightning/trainer/trainer.py @williamfalcon
-/pytorch_lightning/core/hooks.py     @williamfalcon
-/pytorch_lightning/core/lightning.py  @williamfalcon
+pytorch_lightning/callbacks/base.py           @williamfalcon
+pytorch_lightning/core/datamodule.py          @williamfalcon
+pytorch_lightning/trainer/trainer.py          @williamfalcon @tchaton
+pytorch_lightning/core/hooks.py               @williamfalcon
+pytorch_lightning/core/lightning.py           @williamfalcon @tchaton
+pytorch_lightning/trainer/training_loop.py    @tchaton
+pytorch_lightning/trainer/evaluation_loop.py  @tchaton
```
Comment (Borda) [inline .github/CODEOWNERS:22]:
```suggestion
pytorch_lightning/trainer/training_loop.py    @tchaton @SeanNaren
pytorch_lightning/trainer/evaluation_loop.py  @tchaton @SeanNaren
```

**Review by edenlightning — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -5,23 +5,44 @@
...
-/pytorch_lightning/callbacks/base.py  @williamfalcon
-/pytorch_lightning/core/datamodule.py @williamfalcon
-/pytorch_lightning/trainer/trainer.py @williamfalcon
-/pytorch_lightning/core/hooks.py     @williamfalcon
-/pytorch_lightning/core/lightning.py  @williamfalcon
+/pytorch_lightning/callbacks/base.py           @williamfalcon
+/pytorch_lightning/core/datamodule.py          @williamfalcon
+/pytorch_lightning/trainer/trainer.py          @williamfalcon @tchaton
+/pytorch_lightning/core/hooks.py               @williamfalcon
+/pytorch_lightning/core/lightning.py           @williamfalcon @tchaton
```
Comment (tchaton) [inline .github/CODEOWNERS:20]: Can you add me to /pytorch_lightning/core/optimizer.py

**Review by tchaton — APPROVED** [review]

File: .github/CODEOWNERS (hunk 6 above)
Comment (Borda) [reply .github/CODEOWNERS:20]: pls, do it as a suggestion :]

**Comment by Borda** [comment]
@williamFalcon can you review, it also fixes that you have 100% control over this file :]

File: .github/CODEOWNERS (hunk 6 above)
Comment (tchaton) [inline .github/CODEOWNERS:20]:
```suggestion
/pytorch_lightning/core/lightning.py           @williamfalcon @tchaton
/pytorch_lightning/core/optimizer.py           @tchaton
```

**Pull request merged**